### PR TITLE
feat(robot-server): fw version for instrs

### PR DESCRIPTION
**Pull request by sfoster1**
Adds firmware version to the instruments endpoint to make frontend lives easier

### CODE REVIEW

**Comment by smb2268** [comment]
thank you 🤩

File: robot-server/robot_server/instruments/router.py [hunk 1]
```diff
@@ -82,10 +84,13 @@ def _pipette_dict_to_pipette_res(
...
 
 
-def _gripper_dict_to_gripper_res(gripper_dict: GripperDict) -> Gripper:
+def _gripper_dict_to_gripper_res(
+    gripper_dict: GripperDict, fw_version: Optional[int]
+) -> Gripper:
     """Convert GripperDict to Gripper response model."""
     calibration_data = gripper_dict["calibration_offset"]
     return Gripper.construct(
+        firmwareVersion=str(fw_version),
```
Comment (jbleon95) [inline robot-server/robot_server/instruments/router.py]: Do we want to `str` None?

Comment (sfoster1) [reply robot-server/robot_server/instruments/router.py]: Probably not! Good catch

**Review by jbleon95 — APPROVED** [review]
👍

**Comment by sfoster1** [comment]
```
curl -H 'opentrons-version: 3' 10.13.11.96:31950/instruments
{"data": [{"mount": "left", "instrumentType": "pipette", "instrumentModel": "p50_single_v3.4", "serialNumber": "P50SV3420230114A05", "subsystem": "pipette_left", "ok": true, "firmwareVersion": "23", "data": {"channels": 1, "min_volume": 0.5, "max_volume": 50.0, "calibratedOffset": {"offset": {"x": 0.9609375, "y": -2.4609375, "z": 1.2102499999999736}, "source": "user", "last_modified": "2023-07-12T18:48:22.062610+00:00"}}, "instrumentName": "p50_single_flex"}, {"mount": "extension", "instrumentType": "gripper", "instrumentModel": "gripperV1.1", "serialNumber": "GRPV1120230510A07", "subsystem": "gripper", "ok": true, "firmwareVersion": "23", "data": {"jawState": "unhomed", "calibratedOffset": {"offset": {"x": -0.3515625, "y": -0.328125, "z": 0.44800000000001816}, "source": "user", "last_modified": "2023-06-13T20:41:37.372887+00:00"}}}], "meta": {"cursor": 0, "totalLength": 2}}```

**Review by vegano1 — APPROVED** [review]
LGTM

**Pull request merged**